packages/runtime/src/controls/image.ts

```diff
@@ -2,10 +2,25 @@ export type ImageControlData = string
 
 export const ImageControlType = 'makeswift::controls::image'
 
-type ImageControlConfig = { label?: string }
+export const ImageControlValueFormats = {
+  URL: 'URL',
+  WithDimensions: 'WithDimensions',
+} as const
 
-export type ImageControlDefinition = { type: typeof ImageControlType; config: ImageControlConfig }
+type ImageControlValueFormat =
+  typeof ImageControlValueFormats[keyof typeof ImageControlValueFormats]
 
-export function Image(config: ImageControlConfig = {}): ImageControlDefinition {
+type ImageControlConfig = { label?: string; format?: ImageControlValueFormat }
```

Comment:
For simplicity, we could make `format` required here and then default it to `ImageControlFormat.URL` in the factory function.

Reply:
Wait, it means that if they want to provide a config for a label, they also need to add the `format` right?

Reply:
The argument for the `Image` function itself is optional, but in the type for the config it's required. If this is too much of a pain, don't worry about it, though. It would make it so that in our code we always know there's a `format`, though.